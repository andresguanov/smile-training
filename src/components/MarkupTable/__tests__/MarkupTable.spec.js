import { shallowMount } from "@vue/test-utils"
import smMarkupTable from "@components/MarkupTable/MarkupTable.vue"
let wrapper

describe("smMarkupTable", () => {
  //#region Basic render
  it("renders default markup table", () => {
    wrapper = shallowMount(smMarkupTable)
    // expect(wrapper.classes("hover-effect")).toBe(true) is no longer by default
    expect(wrapper.html()).toMatchSnapshot()
  })
  //#endregion

  //#region Props
  it("sets dense", () => {
    wrapper = shallowMount(smMarkupTable, {
      propsData: {
        dense: true
      }
    })
    expect(wrapper.classes("dense")).toBe(true)
  })

  it("sets separator", () => {
    wrapper = shallowMount(smMarkupTable, {
      propsData: {
        separator: "cell",
        separatorColor: "#000"
      }
    })
    expect(wrapper.classes("cell")).toBe(true)
    expect(wrapper.attributes("style")).toBe("--separator-color: #000;")
    expect(wrapper.vm.$options.props.separator.validator("diagonal")).toBeFalsy()
  })
  //#endregion

  //#region Methods
  //#endregion

  //#region Events
  //#endregion

  //#region Slots
  it("renders content", () => {
    const defaultContent = "content"
    wrapper = shallowMount(smMarkupTable, {
      slots: {
        default: defaultContent
      }
    })
    expect(wrapper.text()).toBe(defaultContent)
  })
  //#endregion
})
