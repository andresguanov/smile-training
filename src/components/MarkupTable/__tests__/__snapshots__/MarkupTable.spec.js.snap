// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smMarkupTable renders default markup table 1`] = `<table class="markup-table horizontal" style="--separator-color: #ededed;"></table>`;
